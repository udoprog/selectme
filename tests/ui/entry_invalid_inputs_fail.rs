fn main() {}

#[selectme::main]
fn main_is_not_async() {}

#[selectme::main(foo)]
async fn main_attr_has_unknown_args() {}

#[selectme::main(threadpool::bar)]
async fn main_attr_has_path_args() {}

#[selectme::test]
fn test_is_not_async() {}

#[selectme::test(foo)]
async fn test_attr_has_args() {}

#[selectme::test(foo = 123)]
async fn test_unexpected_attr() {}

#[selectme::test(flavor = 123)]
async fn test_flavor_not_string() {}

#[selectme::test(flavor = "foo")]
async fn test_unknown_flavor() {}

#[selectme::test(flavor = "multi_thread", start_paused = false)]
async fn test_multi_thread_with_start_paused() {}

#[selectme::test(flavor = "multi_thread", worker_threads = "foo")]
async fn test_worker_threads_not_int() {}

#[selectme::test(flavor = "current_thread", worker_threads = 4)]
async fn test_worker_threads_and_current_thread() {}

#[selectme::test]
#[test]
async fn test_has_second_test_attr() {}
