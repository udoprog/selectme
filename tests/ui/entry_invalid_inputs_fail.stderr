error: functions marked with `#[selectme::main]` must be `async`
 --> tests/ui/entry_invalid_inputs_fail.rs:4:1
  |
4 | fn main_is_not_async() {}
  | ^^

error: unknown option `foo`, expected one of: `flavor`, `worker_threads`, `start_paused`
 --> tests/ui/entry_invalid_inputs_fail.rs:6:18
  |
6 | #[selectme::main(foo)]
  |                  ^^^

error: unknown option `threadpool`, expected one of: `flavor`, `worker_threads`, `start_paused`
 --> tests/ui/entry_invalid_inputs_fail.rs:9:18
  |
9 | #[selectme::main(threadpool::bar)]
  |                  ^^^^^^^^^^

error: functions marked with `#[selectme::test]` must be `async`
  --> tests/ui/entry_invalid_inputs_fail.rs:13:1
   |
13 | fn test_is_not_async() {}
   | ^^

error: unknown option `foo`, expected one of: `flavor`, `worker_threads`, `start_paused`
  --> tests/ui/entry_invalid_inputs_fail.rs:15:18
   |
15 | #[selectme::test(foo)]
   |                  ^^^

error: unknown option `foo`, expected one of: `flavor`, `worker_threads`, `start_paused`
  --> tests/ui/entry_invalid_inputs_fail.rs:18:18
   |
18 | #[selectme::test(foo = 123)]
   |                  ^^^

error: no such runtime flavor, the runtime flavors are: "current_thread", "multi_thread"
  --> tests/ui/entry_invalid_inputs_fail.rs:21:27
   |
21 | #[selectme::test(flavor = 123)]
   |                           ^^^

error: no such runtime flavor, the runtime flavors are: "current_thread", "multi_thread"
  --> tests/ui/entry_invalid_inputs_fail.rs:24:27
   |
24 | #[selectme::test(flavor = "foo")]
   |                           ^^^^^

error: the `worker_threads` option requires the "multi_thread" runtime flavor. Use `#[selectme::test(flavor = "multi_thread")]`
  --> tests/ui/entry_invalid_inputs_fail.rs:33:62
   |
33 | #[selectme::test(flavor = "current_thread", worker_threads = 4)]
   |                                                              ^
